src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,100 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+
+    private static final String SPLIT_REGEX = "\\r?\\n";
+    private static final String SEP = ",";
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final String NEW_LINE = System.lineSeparator();
+
+    private static final class Totals {
+        private final int supply;
+        private final int buy;
+
+        Totals(int supply, int buy) {
+            this.supply = supply;
+            this.buy = buy;
+        }
+    }
+
     public void getStatistic(String fromFileName, String toFileName) {
 
+        String content = readFile(fromFileName);
+        Totals totals = parseAndAggregate(content);
+        String report = buildReport(totals);
+        writeFile(toFileName, report);
+    }
+
+    private String readFile(String fromFileName) {
+        try {
+            return Files.readString(Path.of(fromFileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from a file: " + fromFileName + e);
```

This violates the checklist item "Don't ignore exceptions". The current catch block concatenates the exception object into the message instead of preserving it as the cause, which loses the original stack trace and makes debugging harder. Change to rethrow while preserving the cause, for example:
```
catch (IOException e) {
    throw new RuntimeException("Can't read data from file: " + fromFileName, e);
}
```
(See checklist: "Don't ignore exceptions")